src/graph/backend/dnnl/op_executable.hpp

```diff
@@ -2637,6 +2640,195 @@ struct genindex_executable_t : public op_executable_t {
 #endif
 };
 
+struct sdpa_executable_t : public op_executable_t {
+    DECLARE_ARG_INDICES_GETTER;
+
+    sdpa_executable_t(std::shared_ptr<op_t> &op, const dnnl::engine &p_engine,
+            fusion_info_mgr_t &mgr, pd_cache_t &pd_cache) {
+
+        auto md_q = make_dnnl_memory_desc(
+                op->get_input_value(0)->get_logical_tensor());
+        auto md_k = make_dnnl_memory_desc(
+                op->get_input_value(1)->get_logical_tensor());
+        auto md_v = make_dnnl_memory_desc(
+                op->get_input_value(2)->get_logical_tensor());
+        auto md_dst = make_dnnl_memory_desc(
+                op->get_output_value(0)->get_logical_tensor());
+
+        auto scale_dt = impl::data_type::undef;
+        size_t idx = 3;
+        with_scale_ = op->get_attr<bool>(
+                dnnl::impl::graph::dnnl_impl::op_attr::with_scale);
+        if (with_scale_)
+            scale_dt = op->get_input_value(idx++)
+                               ->get_logical_tensor()
+                               .data_type;
+
+        dnnl::memory::desc md_mask;
+        with_mask_ = op->get_attr<bool>(
+                dnnl::impl::graph::dnnl_impl::op_attr::with_mask);
+        if (with_mask_)
+            md_mask = make_dnnl_memory_desc(
+                    op->get_input_value(idx++)->get_logical_tensor());
+
+        dnnl::primitive_attr attr;
+        attr.set_scratchpad_mode(dnnl::scratchpad_mode::user);
+        attr.set_fpmath_mode(
+                static_cast<dnnl::fpmath_mode>(mgr.get_fpmath_mode().mode_));
+        if (op->has_attr(
+                    dnnl::impl::graph::dnnl_impl::op_attr::is_invert_scale))
+            is_invert_scale_ = op->get_attr<bool>(
+                    dnnl::impl::graph::dnnl_impl::op_attr::is_invert_scale);
+
+        is_causal_mask_ = op->get_attr<bool>(
+                dnnl::impl::graph::dnnl_impl::op_attr::with_causal);
```

Comment:
I believe you don't need this long namespace chain.
```suggestion
                op_attr::with_causal);
```